contracts/SwapKiwi.sol

```diff
@@ -326,7 +326,7 @@ contract SwapKiwi is Ownable, ERC721Holder, ERC1155Holder {
     bytes memory _data
   ) internal virtual {
     if (tokenAmount == 0) {
-      IERC721(tokenAddress).safeTransferFrom(from, to, tokenId, _data);
+      IERC721(tokenAddress).transferFrom(from, to, tokenId);
```

Comment:
I think it's same issue for erc1155

Reply:
Yeah, but the standard ERC1155 tokens have not "transferFrom" function. Only have "safeTransferFrom".
Which means every ERC1155 tokens are transferred by "safeTransferFrom" function. 
So if the contract proposes/initiates with ERC1155 tokens, that means the contract has the ERC1155 tokens transferred by "safeTransferFrom" function. Which means the contract should already have "onERC1155Received".

Reply:
onERC1155Received could be malicious. So there should be a single side cancel for that purpose, just in case, even though it will not probably be ever used.

Reply:
Ugh, there is more to that than just adding single side cancel. If single side cancel is available, than people could abuse it as a way not to pay for gas to return nft to other party.

Didn't solidity add some try/catch stuff so you can continue execution if transfer reverts?